aiohttp/connector.py

```diff
@@ -738,6 +741,7 @@ def __init__(
         family: int = 0,
         ssl: Union[None, bool, Fingerprint, SSLContext] = None,
         local_addr: Optional[Tuple[str, int]] = None,
+        network_interface: Optional[str] = None,
```

Comment:
Looking at this, it seems like an overkill to add an extra argument (right?). I'd reuse `local_addr` since it'd be mutually exclusive with the `network_interface` anyway, right?

cURL does the same, it seems — they have an `--interface` CLI option that accepts the interface definition in both formats — name or IP: https://everything.curl.dev/usingcurl/connections/interface.

Reply:
Ah... Looking at dnsmasq, it seems like there's a legit use case for accepting both local address and network interface.
I wonder, though, is it possible to pass an IP address for `network_interface`?

Reply:
The best way would be to have two different arguments. Because `local_addr` requires a tuple of `(ip, port)`. Therefore, I think it's best to let it be.

Reply:
Yes, and it's best not to resolve conversations before the other party sees new comments. It's rather hard to find them when they are hidden, or even know those new comments exist if they are collapsed.

Reply:
Resolving.